### PR TITLE
Proposals generic form, fix #326

### LINKED ISSUE #326: A generic form of a new proposal: stake, title, description
https://github.com/Joystream/apps/pull/355

### PR DESCRIPTION
**Pull request by siman**
@Gamaranto no need to review files under the folder:
`packages/joy-proposals/src/deprecated`

### CODE REVIEW

File: packages/joy-proposals/src/ProposalTypeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { nonEmptyStr } from '@polkadot/joy-utils/index'
+
+type Props = {
+  types: ProposalTypePreviewProps[]
+}
+
+type OnChangeCb = (event: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => void
+
+function containsQuery(text?: string, query?: string): boolean {
+  return text && query && nonEmptyStr(text) && nonEmptyStr(query)
```
Comment (Gamaranto) [inline packages/joy-proposals/src/ProposalTypeSelector.tsx:14]: You can shorten this to  
```javascript
nonEmptyStr(text) && nonEmptyStr(query) ? text.toLowerCase().includes(query) : true 
```

since ```nonEmptyStr(value)``` will check the type of ```value``` and return false in the case of `undefined` or `null`

Comment (siman) [reply packages/joy-proposals/src/ProposalTypeSelector.tsx:14]: If I do it, then TypeScript is not happy :(
image

Comment (siman) [reply packages/joy-proposals/src/ProposalTypeSelector.tsx:14]: TS cannot see that that `nonEmptyStr()` checks that `text` and `query` are not `undefined`.
